overlord/snapstate/handlers.go

```diff
@@ -819,3 +819,194 @@ func (m *SnapManager) doDiscardSnap(t *state.Task, _ *tomb.Tomb) error {
 	Set(st, snapsup.Name(), snapst)
 	return nil
 }
+
+// aliases v2
```

Comment:
It would be great to have here a comment explaining why we have the multi-step process below with multiple tasks, and why we need the pending flag, as this will definitely feel a bit surprising and confusing on a first read.

It's okay to do that on a follow up though.